library.c

```diff
@@ -2371,6 +2441,36 @@ redis_unpack(RedisSock *redis_sock, const char *val, int val_len, zval *z_ret)
                     return 1;
                 }
             }
+#endif
+            break;
+        case REDIS_COMPRESSION_LZ4:
+#ifdef HAVE_REDIS_LZ4
+            {
+                char *data;
+                int size, res;
+                long offset;
+
+                if (*val != '\0') {
+                    break;
+                }
+
+                offset = sizeof(int);
+                memcpy(&size, val + 1, offset);
+                offset++;
+
+                data = emalloc(size + 1);
+                res = LZ4_decompress_safe(val + offset, data, val_len - offset, size);
+                if (res > 0) {
+                    if (redis_unserialize(redis_sock, data, size, z_ret) == 0) {
```

Comment:
You don't need the else block here.  Otherwise, it doesn't deserialize and leaks memory.

Reply:
adjusted, thanks